common/context.go

```diff
@@ -0,0 +1,13 @@
+package common
+
+import "context"
+
+// WasContextClosed will return true if the provided context has been closed
+func WasContextClosed(ctx context.Context) bool {
```

Comment:
I would rename it to `IsContextDone` because:
1. the context can also be canceled in case of timeout or cancel(), not just closed
2. Was -> Is because the context can be closed/canceled when within the function so putting the action into the present is more meaningful.

Reply:
also, please guard the `ctx` for nil values. the code would panic. I'd say that you should return true if it is nil

Reply:
done, thanks